ANDROID: memcg: swap handling on Android

Android devices typically do not support `swapoff -a` or `swapon -a`
globally. Modify the memcg regression tests to parse `/proc/swaps` and
turn off/on individual swap devices sequentially.

Test: atest vts_ltp_test_arm:controllers.memcg_regression_32bit

diff --git a/testcases/kernel/controllers/memcg/regression/memcg_regression_test.sh b/testcases/kernel/controllers/memcg/regression/memcg_regression_test.sh
index 734d803d5..6e259fb15 100755
--- a/testcases/kernel/controllers/memcg/regression/memcg_regression_test.sh
+++ b/testcases/kernel/controllers/memcg/regression/memcg_regression_test.sh
@@ -221,6 +221,7 @@
 	local test_path
 	test_path="$test_dir/0"
 	create_testpath "$test_path"
+	swaps=$(cut -f1 /proc/swaps | tail -n +2)
 
 	./memcg_test_4.sh "$cgroup_version" "$mount_point" "$test_path"
 
@@ -236,7 +237,9 @@
 	# if test_4.sh gets killed, it won't clean cgroup it created
 	cleanup_testpath "$test_path"
 
-	swapon -a
+	for s in $swaps; do
+		swapon $s
+	done
 }
 
 . cgroup_lib.sh
diff --git a/testcases/kernel/controllers/memcg/regression/memcg_test_4.sh b/testcases/kernel/controllers/memcg/regression/memcg_test_4.sh
index 734d803d5..6e259fb15 100755
--- a/testcases/kernel/controllers/memcg/regression/memcg_test_4.sh
+++ b/testcases/kernel/controllers/memcg/regression/memcg_test_4.sh
@@ -29,7 +29,9 @@
 echo 40M > "$test_path/$memory_limit"
 
 # turn off swap, and swapoff will be killed
-swapoff -a
+for s in $(cut -f1 /proc/swaps | tail -n +2); do
+	swapoff $s
+done
 sleep 1
 echo $pid > "$mount_point/$task_list" 2> /dev/null
 echo $$ > "$mount_point/$task_list"  2> /dev/null
